### PR TITLE
Fix test failures on big-endian systems

### PR DESCRIPTION
**Pull request by QuLogic**
Essentially, the test data is little-endian, but the test setup is imprecise and uses native endianness. This causes failures when checking results. See commit messages for more information.

* [x] Unit tests and/or doctests in docstrings
* [x] ``tox -e py37`` passes locally
* [?] ``tox -e py27`` passes locally
* [N/A] Docstrings and API docs for any new/modified user-facing classes and functions
* [ ] Changes documented in docs/release.rst
* [N/A] ``tox -e docs`` passes locally
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage to 100% (Coveralls passes)

### CODE REVIEW

File: numcodecs/tests/test_astype.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def test_backwards_compatibility():
     # integers
     arrs = [
         np.arange(1000, dtype='<i4'),
-        np.random.randint(0, 200, size=1000, dtype='<i4').reshape(100, 10, order='F'),
+        np.random.randint(0, 200, size=1000, dtype='i4').astype('<i4').reshape(100, 10, order='F'),
```
Comment (alimanfoo) [inline numcodecs/tests/test_astype.py]:
```suggestion
        np.random.randint(0, 200, size=1000, dtype='<i4').reshape(100, 10, order='F'),
```

Could simplify.

File: numcodecs/tests/test_delta.py [hunk 2]
```diff
@@ -16,10 +16,10 @@
...
 # mix of orders: C, F
 arrays = [
-    np.arange(1000, dtype='i4'),
-    np.linspace(1000, 1001, 1000, dtype='f4').reshape(100, 10),
-    np.random.normal(loc=1000, scale=1, size=(10, 10, 10)).astype('f8'),
-    np.random.randint(0, 200, size=1000, dtype='u2').reshape(100, 10, order='F'),
+    np.arange(1000, dtype='<i4'),
+    np.linspace(1000, 1001, 1000, dtype='<f4').reshape(100, 10),
+    np.random.normal(loc=1000, scale=1, size=(10, 10, 10)).astype('<f8'),
+    np.random.randint(0, 200, size=1000, dtype='u2').astype('<u2').reshape(100, 10, order='F'),
```
Comment (alimanfoo) [inline numcodecs/tests/test_delta.py]:
```suggestion
    np.random.randint(0, 200, size=1000, dtype='<u2').reshape(100, 10, order='F'),
```

Can it be simplified like this?

File: numcodecs/tests/test_astype.py (hunk 1 above)
Comment (QuLogic) [reply numcodecs/tests/test_astype.py]: No, see the commit message; `np.random.randint` does not accept non-native byteorder.

**Comment by alimanfoo** [comment]
Just to say I'm about to go offline for a couple of weeks so apologies if I go quiet, hopefully someone else in @zarr-developers/core-devs might be able to help but if not more than happy to pick up again when I'm back.

File: numcodecs/tests/test_pickles.py [hunk 3]
```diff
@@ -54,5 +55,7 @@ def test_repr():
     check_repr("Pickle(protocol=-1)")
 
 
+@pytest.mark.skipif(sys.byteorder != 'little',
+                    reason='Pickle does not restore byte orders')
```
Comment (jakirkham) [inline numcodecs/tests/test_pickles.py]: Is there any more information about why this is?

**Comment by jakirkham** [comment]
@alimanfoo, could you please recheck and let us know if your concerns are addressed?

File: numcodecs/tests/test_pickles.py (hunk 3 above)
Comment (QuLogic) [reply numcodecs/tests/test_pickles.py]: No, not much specifically; all I can say is this: https://github.com/zarr-developers/numcodecs/issues/181#issuecomment-477507464

**Comment by jakirkham** [comment]
Going to go ahead and merge this as it at least improves the situation. Though more improvements to this would be welcome in subsequent PRs.

**Pull request merged**